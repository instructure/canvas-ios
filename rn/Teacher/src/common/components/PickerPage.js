//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import React, { Component } from 'react'
import {
  FlatList,
  Image,
} from 'react-native'
import Screen from '../../routing/Screen'
import { createStyleSheet } from '../stylesheet'
import icon from '../../images/inst-icons'
import Row from './rows/Row'

type Props = {
  onSelect: (string) => any,
  options: { [string]: string },
  selectedValue: string,
  title: string,
}

export default class PickerPage extends Component<Props> {
  render () {
    const { onSelect, options, selectedValue, title } = this.props
    return (
      <Screen
        title={title}
        drawUnderNavBar
      >
        <FlatList
          style={styles.container}
          data={Object.keys(options)}
          renderItem={({ item: key }) =>
            <Row
              key={key}
              style={styles.row}
              border='bottom'
              title={options[key]}
              titleStyles={{ fontWeight: 'normal' }}
              onPress={() => onSelect(key)}
              accessories={
                key === selectedValue &&
                <Image
                  style={styles.check}
                  source={icon('check', 'solid')}
                />
              }
              accessibilityState={{ selected: (key === selectedValue) }}
            />
          }
        />
      </Screen>
    )
  }
}

const styles = createStyleSheet((colors, vars) => ({
  container: {
    backgroundColor: colors.backgroundGrouped,
  },
  row: {
    backgroundColor: colors.backgroundGroupedCell,
  },
  check: {
    tintColor: colors.textSuccess,
    height: 20,
    width: 20,
    resizeMode: 'contain',
    marginLeft: vars.padding,
  },
}))
