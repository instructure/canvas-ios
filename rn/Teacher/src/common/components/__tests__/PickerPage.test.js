//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow
import React from 'react'
import { shallow } from 'enzyme'

import PickerPage from '../PickerPage'

describe('PickerPage', () => {
  let props
  beforeEach(() => {
    props = {
      onSelect: jest.fn(),
      title: 'Pick Something',
      selectedValue: 'b',
      options: {
        a: 'First One',
        b: 'Second One',
        c: 'Third One',
      },
    }
  })

  it('renders the given options as rows', () => {
    const tree = shallow(<PickerPage {...props} />)
    expect(tree).toMatchSnapshot()
    for (const key of Object.keys(props.options)) {
      const row = shallow(tree.find('FlatList').prop('renderItem')({ item: key }))
      expect(row).toMatchSnapshot(key)
    }
  })

  it('calls the onSelect prop with a row is clicked', () => {
    const tree = shallow(<PickerPage {...props} />)
    const row = shallow(tree.find('FlatList').prop('renderItem')({ item: 'a' }))
    row.simulate('Press')
    expect(props.onSelect).toHaveBeenCalledWith('a')
  })
})
