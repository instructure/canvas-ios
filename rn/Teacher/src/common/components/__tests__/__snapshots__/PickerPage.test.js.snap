// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerPage renders the given options as rows 1`] = `
<Screen
  drawUnderNavBar={true}
  navBarStyle="modal"
  showDismissButton={true}
  title="Pick Something"
>
  <FlatList
    data={
      Array [
        "a",
        "b",
        "c",
      ]
    }
    renderItem={[Function]}
    style={
      Object {
        "backgroundColor": "#F5F5F5",
      }
    }
  />
</Screen>
`;

exports[`PickerPage renders the given options as rows: a 1`] = `
<ForwardRef
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  onPress={[Function]}
  style={
    Array [
      undefined,
      Object {
        "borderBottomColor": "#C7CDD1",
        "borderBottomWidth": 0.5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 54,
          "paddingBottom": 12,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
          Object {
            "marginLeft": 0,
          },
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={0}
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontWeight": "600",
            },
            Object {
              "fontWeight": "normal",
            },
          ]
        }
      >
        First One
      </Text>
    </View>
  </View>
</ForwardRef>
`;

exports[`PickerPage renders the given options as rows: b 1`] = `
<ForwardRef
  accessibilityTraits={
    Array [
      "button",
      "selected",
    ]
  }
  onPress={[Function]}
  style={
    Array [
      undefined,
      Object {
        "borderBottomColor": "#C7CDD1",
        "borderBottomWidth": 0.5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 54,
          "paddingBottom": 12,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
          Object {
            "marginLeft": 0,
          },
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={0}
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontWeight": "600",
            },
            Object {
              "fontWeight": "normal",
            },
          ]
        }
      >
        Second One
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "checkSolid",
            }
          }
          style={
            Object {
              "height": 20,
              "marginLeft": 16,
              "resizeMode": "contain",
              "tintColor": "#00AC18",
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
</ForwardRef>
`;

exports[`PickerPage renders the given options as rows: c 1`] = `
<ForwardRef
  accessibilityTraits={
    Array [
      "button",
    ]
  }
  onPress={[Function]}
  style={
    Array [
      undefined,
      Object {
        "borderBottomColor": "#C7CDD1",
        "borderBottomWidth": 0.5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "flexDirection": "row",
          "minHeight": 54,
          "paddingBottom": 12,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          },
          Object {
            "marginLeft": 0,
          },
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={0}
        style={
          Array [
            Object {
              "color": "#2D3B45",
              "fontWeight": "600",
            },
            Object {
              "fontWeight": "normal",
            },
          ]
        }
      >
        Third One
      </Text>
    </View>
  </View>
</ForwardRef>
`;
